### PR TITLE
Integrating data

### LINKED ISSUE #390: Missing content - data integration
On this read-through, i realize we don't have anything about integrating different datasets (whether that is admin to survey data, different rounds/levels of survey data, linking "big" data, etc). i think this is a significant gap, especially since we shifted the focus from primary data, and one we should address. I'm not sure where though. some could go in chapter 2, then details in 5/6. it would fit in 6 if we expand the scope of the section on 'construction' to include dataset construction rather than just indicators. a more dramatic option would be to split the survey-specific content in chapter 5 to its own chapter and keep this with the other content on data acquisition. 

some points i think this would include:
Ch 2: importance of understanding unit of response and identifier for all the datasets you work with, thinking about different levels on which they could be integrated

Ch 5/6: why merging can be complex in practice and what to do about it
- merging datasets at the same unit of response **with** consistent numeric ID (i.e. baseline to endline) -- this is aligned with the iedup content we already have. 
- merging datasets at same unit of response with string ID (i.e. name) and guidance on how to do fuzzy merges
- merging datasets at same unit of response without a shared ID variable (e.g. admin data to survey data) 
- merging datasets at different units of response - e.g. spatial matching

@bbdaniels @kbjarkefur @luizaandrade -- thoughts?

### PR DESCRIPTION
**Pull request by bbdaniels**
Closes #390.

### CODE REVIEW

File: chapters/data-analysis.tex [hunk 1]
```diff
@@ -370,23 +370,15 @@ \subsection{Documenting data cleaning}
...
 
-\section{Constructing final indicators}
+\section{Constructing analysis datasets}
 
 % What is construction -------------------------------------
-The third stage is construction of the variables of interest for analysis.
-It is at this stage that the raw data is transformed into analysis data.
-This is done by creating derived variables (dummies, indices, and interactions, to name a few),
+The third stage is construction of the dataset you will use for analysis.
+It is at this stage that the raw data is transformed into analysis-ready data,
```
Comment (kbjarkefur) [inline chapters/data-analysis.tex]:
```suggestion
It is at this stage that the clean data is transformed into analysis-ready data, 
```
Data cleaning starts at raw data, variable construction starts with clean data, right?

File: chapters/data-analysis.tex [hunk 2]
```diff
@@ -442,18 +433,27 @@ \section{Constructing final indicators}
...
-but that use an external or string ID such as a name,
-which do not always perfectly match.
-In these cases, you will need to extensively analyse the merging patterns
+To do this, you will need to consider the unit of response for each dataset, 
+and the identifying variable, to understand how they can be merged.
+
+If the datasets you need to join have the same unit of response, 
+merging may be straightforward.
+The simplest case is merging datasets at the same unit of response 
+which use a consistent numeric ID.
```
Comment (kbjarkefur) [inline chapters/data-analysis.tex]:
```suggestion
which use a consistent, uniquely and fully identifying ID variable. 
```
We have link to what _uniquely and fully identifying_ means earlier in this chapter

File: chapters/data-analysis.tex [hunk 3]
```diff
@@ -425,67 +416,113 @@ \section{Constructing final indicators}
...
+using a spatial match,
+or combining school administrative data, such as attendance records and test scores,
+with household demographic characteristics from a survey.
+Sometimes these cases are conceptually straightforward.
+For example, merging a dataset of health care providers
+with a dataset of patients comes with a clear linking relation between the two;
+the challenge usually occurs in correctly defining statistical aggregations
+if the merge is intended to result in a dataset at the provider level.
+However, other cases may not be designed with the intention to be merged together,
+such as a dataset of infrastructure access points such as water pumps or schools
```
Comment (kbjarkefur) [inline chapters/data-analysis.tex]:
```suggestion
such as a dataset of infrastructure access points, for example, water pumps or schools
```
Two _such as_ in the same sentence

File: chapters/data-analysis.tex [hunk 4]
```diff
@@ -425,67 +416,113 @@ \section{Constructing final indicators}
...
+and are extensively documented separately from other data construction tasks.
+
+Integrating different datasets may involve changing the structure of the data,
+e.g. changing the unit of observation through collapses or reshapes.
+This should always be done with great care. 
+Two issues to pay extra attention to are missing values and dropped observations.
+Merging, reshaping and aggregating data sets can change both the total number of observations
+and the number of observations with missing values.
+Make sure to read about how each command treats missing observations and,
+whenever possible, add automated checks in the script that throw an error message if the result is changing.
```
Comment (kbjarkefur) [inline chapters/data-analysis.tex]:
```suggestion
whenever possible, add automated checks in the script that throw an error message if the result is different than what you expect.
```
How to handle _change_ in unit of obs is what we are talking about here

**Comment by bbdaniels** [comment]
This is good from me.

**Pull request merged**